### ISSUE TITLE
Validation accuracy is not changing

### ISSUE THREAD

**Issue by usmaann** [amaiya/ktrain#201]
Hi.... 
I am using Ktrain for text classification task.  I observed that validation accuracy is not changing ....its the same..please advise what could be the reason?

Piece of code:

`MODEL_NAME = 'albert-large-v2'
t = text.Transformer(MODEL_NAME, maxlen=500, class_names=train_df.tweet_label_int.unique())
trn = t.preprocess_train(x_train, y_train)
val = t.preprocess_test(x_test, y_test)
model = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=6)`

`# learner.fit_onecycle(8e-5, 20) 

Below is the image showing same validation accuracy for all epochs
![image](https://user-images.githubusercontent.com/38275359/87770562-83b13380-c862-11ea-8519-b321fc045c08.png)

I feel I have to change something in optimizer or activation function? but don't have any idea from where I can and how? appreciate if you can guide considering me a newbie

PS: Running on google colab- using (!pip3 install ktrain)

**Comment by amaiya** [comment]
It is probably something related to your data and/or labels, rather than the code.  If you post the data or a Google Colab reproducing the problem, I can take a look.

**Comment by usmaann** [comment]
Thanks alot for your quick reply.


[test_sample-x.xlsx](https://github.com/amaiya/ktrain/files/4937813/test_sample-x.xlsx)
[train_sample-x.xlsx](https://github.com/amaiya/ktrain/files/4937814/train_sample-x.xlsx)

Sample of Train and Test data is attached. Could you please advise? I use batch size= 6, lr=8e-5

**Comment by usmaann** [comment]
> It is probably something related to your data and/or labels, rather than the code. If you post the data or a Google Colab reproducing the problem, I can take a look.

Thanks alot for your quick reply.
[test_sample-x.xlsx](https://github.com/amaiya/ktrain/files/4937820/test_sample-x.xlsx)
[train_sample-x.xlsx](https://github.com/amaiya/ktrain/files/4937821/train_sample-x.xlsx)


Sample of Train and Test data is attached. Could you please advise? I use batch size= 6, lr=8e-5

**Comment by amaiya** [comment]
What are the labels supposed to represent?

**Comment by usmaann** [comment]
> What are the labels supposed to represent?

 text is critical of vaccines then the label is 1; otherwise the tweet is vaccine promoting or neutral then the label is 0

**Comment by amaiya** [comment]
I think the problem is that the applied labels are inconsistent and don't always make sense.  For instance, this tweet is labeled as 1 (critical of vaccines):
> In encouraging sign, Ebola vaccine appears to provide long-lasting protection https://t.co/sa5V7ppB43 via @statnews

**Comment by usmaann** [comment]
Used same data when I used Keras for same transformer model... it is working fine...so what could be reason? seems no issue with data?

**Comment by amaiya** [comment]
I tried your dataset with **DistilBert**  in addition to other models (e.g., **BERT**, **XLNet**) and **every model I selected trained just fine and yielded a low loss** (despite aforementioned issues with your data), which suggests the problem isn't with *ktrain* but with the model (see below).  It is only when you use `albert-large-v2`  that the loss doesn't improve with the hyperparameters you used.   Also, *ktrain* is just a lightweight wrapper around `tf.keras`, so it is very unlikely that something would work in `tf.keras` and not in *ktrain*.

When you used `keras` and `albert-large-v2` did you use the `transformers` library? From your Colab, it looks like you were using TF HUB, so perhaps this is a bug in `transformers` library with the Albert model?  You might also try different hyperparameters lke increasing the batch size.  In any case, things do work with other models like DistilBERT (which is fast and light like Albert should you choose to use it).

```python
import pandas as pd
df = pd.read_csv('data/example.csv')
y_train = df['tweet_label_int'].values
x_train = df['Tweets'].values

# build, train, and validate model (Transformer is wrapper around transformers library)
import ktrain
from ktrain import text
MODEL_NAME = 'distilbert-base-uncased'
#MODEL_NAME = 'albert-large-v2'
t = text.Transformer(MODEL_NAME, maxlen=128, class_names=['not critical', 'critical'])
trn = t.preprocess_train(x_train, y_train)
val = None
model = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=8)
learner.fit_onecycle(5e-5, 6)
```

```
Begin training using onecycle policy with max lr of 5e-05...
Epoch 1/6
13/13 [==============================] - 1s 60ms/step - loss: 0.6623 - accuracy: 0.6364
Epoch 2/6
13/13 [==============================] - 1s 58ms/step - loss: 0.5182 - accuracy: 0.8182
Epoch 3/6
13/13 [==============================] - 1s 57ms/step - loss: 0.4607 - accuracy: 0.8182
Epoch 4/6
13/13 [==============================] - 1s 59ms/step - loss: 0.3797 - accuracy: 0.8182
Epoch 5/6
13/13 [==============================] - 1s 58ms/step - loss: 0.2312 - accuracy: 0.8586
Epoch 6/6
13/13 [==============================] - 1s 58ms/step - loss: 0.1579 - accuracy: 0.9293
```

**Comment by usmaann** [comment]
> albert-large-v2

Thank you very much ... Could you please how to handle highly imbalanced text data and use transformers via ktrains? any tutorial/ notebook? Appreciate that.

**Comment by amaiya** [comment]
Imbalanced data is handled in the same way as in `tf.keras`:  all `*fit*` methods accept a `class_weight` parameter.  See [this post](https://datascience.stackexchange.com/questions/13490/how-to-set-class-weights-for-imbalanced-classes-in-keras).

**Comment by usmaann** [comment]
> Imbalanced data is handled in the same way as in `tf.keras`: all `*fit*` methods accept a `class_weight` parameter. See [this post](https://datascience.stackexchange.com/questions/13490/how-to-set-class-weights-for-imbalanced-classes-in-keras).

Thanks for your quick response (Appreciate that).

in my training data:  Class ''0'' = 7377 samples  and Class"1"= 974 samples. (Class '0' is almost 7.5 (8) times more than class '1')

now as you mentioned ...What iI should be doing from below mentioned examples?

1)

`class_weight = {0: 1,
                1: 8}`  # not sure if I am doing right or not...Pls advise

`learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=6, class_weight=class_weights)`


or 

`class_weight = {0: 1,
                1: 8}`   # not sure if I am doing right or not...Pls advise

`learner.autofit(5e-5)`

**Comment by usmaann** [comment]
Hi....

In case we want to use ALBERT models do we have to use any special "preprocess_mode" or default standard will work? Also what if we want to use other variant of BERT? in below it downloads only one variant what if we want to use another one?

Appeciate if you could clear this query too. Thanks

![image](https://user-images.githubusercontent.com/38275359/88616600-e8c51e80-d0d7-11ea-9bce-fa45459fce65.png)
